themes/absimple/absimple.theme.sh

```diff
@@ -0,0 +1,384 @@
+#!/usr/bin/env bash
+# vim: ft=bash ts=2 sw=2 sts=2
+#
+# absimple theme 
+# an agnoster fork for bash
+#
+# tested on termux at android 2021-01-03
+#
+# the aim of this theme is to only show you *relevant* information.
+# ie: it will only show git information when in a git working directory.
+# also, everything from the current user and hostname to whether the last call exited with an error or whether background jobs are running in this shell will all be displayed automatically when appropriate.
+
+PROMPT_DIRTRIM=2 # bash4 and above
+
+######################################################################
+DEBUG=0
+debug() {
+    if [[ ${DEBUG} -ne 0 ]]; then
+        >&2 echo -e $*
+    fi
+}
+
+######################################################################
+### Segment drawing
+# A few utility functions to make it easy and re-usable to draw segmented prompts
+
+CURRENT_BG='NONE'
+CURRENT_RBG='NONE'
+SEGMENT_SEPARATOR=''
+RIGHT_SEPARATOR=''
+LEFT_SUBSEG=''
+RIGHT_SUBSEG=''
+
+text_effect() {
+    case "$1" in
+        reset)      echo 0;;
+        bold)       echo 1;;
+        underline)  echo 4;;
+    esac
+}
+
+# to add colors, see
+# http://bitmote.com/index.php?post/2012/11/19/Using-ANSI-Color-Codes-to-Colorize-Your-Bash-Prompt-on-Linux
+# under the "256 (8-bit) Colors" section, and follow the example for orange below
+fg_color() {
+    case "$1" in
+        black)      echo 30;;
+        red)        echo 31;;
+        green)      echo 32;;
+        yellow)     echo 33;;
+        blue)       echo 34;;
+        magenta)    echo 35;;
+        cyan)       echo 36;;
+        white)      echo 37;;
+        orange)     echo 38\;5\;166;;
+    esac
+}
+
+bg_color() {
+    case "$1" in
+        black)      echo 40;;
+        red)        echo 41;;
+        green)      echo 42;;
+        yellow)     echo 43;;
+        blue)       echo 44;;
+        magenta)    echo 45;;
+        cyan)       echo 46;;
+        white)      echo 47;;
+        orange)     echo 48\;5\;166;;
+    esac;
+}
+
+# TIL: declare is global not local, so best use a different name
+# for codes (mycodes) as otherwise it'll clobber the original.
+# this changes from BASH v3 to BASH v4.
+ansi() {
+    local seq
+    declare -a mycodes=("${!1}")
```

Comment:
Maybe this is the problem of the original `agnoster`, but `local` and `declare` are equivalent in function scopes, so we want to use either one consistently across the codebase. Basically, we want to rewrite all of `declare` in the function scopes to `local`. Also, the variable `i` is not declared properly here.

There are also similar cases in other lines and functions.

I would appreciate it if you could also check whether there are more variables that can be made `local`.

Could you also update `agnoster`?

Reply:
yes. i didn't really modify much of agnoster.

and i can't update agnoster.

in fact, from your review of this file alone, something tells me i should even abandon absimple... at least for 2022. i can feel it got too complex to me!

i can give it a shot to go for `local` right now, though...